### ISSUE TITLE
请问没有使用正常的 Conv layer 而不是 Deformable 的测试结果呢？

### ISSUE THREAD

**Issue by kzjeef** [xingyizhou/CenterNet#109]
Hi，

CenterNet 的速度和精度都非常棒， 先点个赞。

由于 Deformable 卷积在代码实现层面无法使用例如 TensorCore 之类的加速指令， 所以在实际部署中， 速度是没有普通的Conv 运行的快的。 所以我想请教下 @Duankaiwen 有没有使用普通的卷积 OP把 CenterNet 的框架结合起来的想法？

不知道精度和速度会是怎么样的。

谢谢。

**Comment by cwjhx** [comment]
--arch dlav0_34
这个没有使用Deformable

**Comment by xingyizhou** [comment]
I decided to delay to respond to this issue because you @ the wrong author.

**Comment by XinjieInformatik** [comment]
hi @xingyizhou, thanks for your great work! I also have the same confusion. Have you test the precision of a normal Conv in your network?

**Comment by OPPOA113** [comment]
@xingyizhou @cwjhx 
--arch dlav0_34这个网络结构的，我测试了。dlav0_34精度上在我自己的数据集效果和dla_dcn的相差不大，但是速度上差距较大。单单是从demo.py跑出的输出的|net xxx s这一栏来看，dla_dcn的约为20ms左右，而dlav0_34的基本都在100ms以上。所以想问下是否因为dcn采用c编译的原因，产生这么大的差距？
另外，dlav0的模型的网络结构，貌似是有点问题。在level3.Tree.project的bn之后就没有处理，而多出一个conv，这是否会影响模型结构呢，。。
![level3 project](https://user-images.githubusercontent.com/4974108/58782192-7ca07500-8610-11e9-833e-f4467bae4381.jpg)

**Comment by kzjeef** [comment]
> I decided to delay to respond to this issue because you @ the wrong author.

:D, My bad, Sorry for than, 
I send this question to wrong question to another CenterNet before, and copy that question to here.. 
How stupid I am !

**Comment by wangshankun** [comment]
> @xingyizhou @cwjhx
> --arch dlav0_34这个网络结构的，我测试了。dlav0_34精度上在我自己的数据集效果和dla_dcn的相差不大。。
> 另外，dlav0的模型的网络结构，貌似是有点问题。在level3.Tree.project的bn之后就没有处理，而多出一个conv，这是否会影响模型结构呢，。。
> ![level3 project](https://user-images.githubusercontent.com/4974108/58782192-7ca07500-8610-11e9-833e-f4467bae4381.jpg)

项目中小数据集确实影响不大，不知道有没有coco的测试结果

**Comment by hexiangquan** [comment]
@OPPOA113 you have convert model to caffe  model ?

**Comment by OPPOA113** [comment]
@wangshankun 不好意思没有测试coco数据集。
@hexiangquan 我的电脑上没有转成功，不过我知道有人转成功了的。我的那个图就是用dlav0转出来用netscope查看的。可以看到，网络结构中间有两个卷积输出是没有用到的，还有一个我没有截图出来。。而这个prototxt是用pytorch1.0版本转出来，而我用0.4版本则转不成功，所以我觉得可能是版本的原因。
@xingyizhou 另外我看到的没用到的两个conv，请大神确认下是否是这样。谢谢

**Comment by hexiangquan** [comment]
@OPPOA113 i have converted to caffe  ,get the same result .have two conv used.

**Comment by hexiangquan** [comment]
i have get right result in caffe  .

**Comment by OPPOA113** [comment]
@hexiangquan Nice works!
是dlav0模型吗，中间的两个没用到的conv卷积影响检测结果吗？

**Comment by hexiangquan** [comment]
@OPPOA113  i have compare torch's output and caffe's ,the same value

**Comment by OPPOA113** [comment]
@hexiangquan 是dlav0 的模型吗？
中间没有到的两个输出卷积如何处理的？
谢谢

**Comment by hexiangquan** [comment]
@OPPOA113 just ignore

**Comment by zzzxxxttt** [comment]
Here are my experiment results:
COCO
resdcn-18 => 28.0/30.0/33.3
resnet-18 => 23.2/24.7/30.6

PascalVOC@384x384
resdcn-18 => 70.41/72.39
resnet-18 => 67.65/70.21

PascalVOC@512x512
resdcn-18 => 72.84/75.39
resnet-18 => 60.62/62.71

It is pretty strange that resnet-18 at resolution 512x512 perform a lot worse than at 384x384.

**Comment by xingyizhou** [comment]
Thanks for the discussion. My results for `--arch dlav0_34` is 31.8/ 33.4 COCO AP (90/ 50fps), for non-flip/ flip test. Note that dlav0 differs from the dla used in the main experiments in two aspect: with/ without dcn and with/ without lower-layer upsample (see the figure in the supplement). We haven't tried paper-dla without dcn. I am not aware of the not-used conv layers issue.

**Comment by NEU-Gou** [comment]
Just a data point for someone interesting:
Due to the layer definition inconsistent in PyTorch and tf.keras,  I substitute Conv2DTranspose layer when `--arch res_18` with Upsample('nearest') + Conv2D. After training with default optimizer setting, 
COCO @ 512x512
24.9/26.3 w/o or with flip
Result for the original architecture is 24.4 w/o flip.

**Comment by Fanpanda** [comment]
> Here are my experiment results:
> COCO
> resdcn-18 => 28.0/30.0/33.3
> resnet-18 => 23.2/24.7/30.6
> 
> PascalVOC@384x384
> resdcn-18 => 70.41/72.39
> resnet-18 => 67.65/70.21
> 
> PascalVOC@512x512
> resdcn-18 => 72.84/75.39
> resnet-18 => 60.62/62.71
> 
> It is pretty strange that resnet-18 at resolution 512x512 performs a lot worse than 384x384.

I also did the same experiment, but I only try the model without dcn. And my result on PascalVOC@512x512 is 71.97 (resnet-18).  I watch the loss is strange on val set, which raise after 60th epoch.  Did you meet similar problems? Thanks.

**Comment by zhliu2018** [comment]
> Result for the original architecture is 24.4 w/o flip.

Result for the original architecture is 28.1 /30.0 w/o flip in author's model zoo

**Comment by dl8207531** [comment]
@zzzxxxttt @Fanpanda 你们能分享下训练日志吗？ 我想看下loss过程 
我现在用resnet18在coco下只有16

**Comment by macqueen09** [comment]
@Fanpanda can you give your code how to train a model without DCN . 

by the way. loss raise after some epoch may be normal in [this issue](https://github.com/xingyizhou/CenterNet/issues/148)
